### PR TITLE
Factor out character decoder from Pipe class.

### PR DESCRIPTION
**Pull request by danfuzz**
This will make it reusable in some upcoming additions.

### CODE REVIEW

**Comment by danfuzz** [comment]
/cc @mikefleming 
R=mike

File: lib/decoder.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * Helper functions
+ */
+
+/**
+ * Gets whether the named encoding is considered valid as an argument
+ * to `Stream.setEncoding()`.
+ */
+function isValid(encodingName) {
+    switch (encodingName) {
+        case undefined:
```
Comment (davidbyttow) [inline lib/decoder.js:26]: oh?

Comment (danfuzz) [inline lib/decoder.js:26]: This works and I believe is correct per-spec.

**Comment by mikefleming** [comment]
LGTM

I think I'll get in the habit of using Object.freeze after my class definitions too.

**Comment by danfuzz** [comment]
Thanks!

And re: freeze(), yeah, it seems like a pretty sane thing to do to me.

**Pull request merged**